src/Tempest/Database/src/Config/DatabaseConfig.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Config;
+
+use Tempest\Database\Tables\NamingStrategy;
+
+interface DatabaseConfig
+{
+    public string $dsn { get; }
+
+    public NamingStrategy $namingStrategy { get; }
+
+    public DatabaseDialect $dialect { get; }
```

Comment:
Do we still need the dialect? My understanding is that it will be a one-to-one mapping between the config instance and the dialect. Maybe we should simply have a mapping somewhere?

Reply:
We do need the DatabaseDialect of some sorts somewhere, as it will change how queries are build. We can move it outside the DatabaseConfig if you want. But honestly, I don't think it will be "better". Currently if you want to change the databaseConfig (I don't know why), you can, if we hard map the DatabaseDialect to our known DatabaseConfig children one cannot create their own config without modifying the dialect mapper as well. I seems a bit weird to me to introduce this lock-in.

Reply:
> I don't know why

That was my reasoning: if we'll never change it, why keep it here?

OTOH, as you say: 

> I seems a bit weird to me to introduce this lock-in.

That's also a good point. It's not really in the way, so I'm fine keeping it as is.